worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -21,6 +21,10 @@ import java.nio.ByteBuffer
 import java.util.concurrent.{ConcurrentHashMap, ThreadPoolExecutor}
 import java.util.concurrent.atomic.{AtomicBoolean, AtomicIntegerArray}
 
+import scala.concurrent.ExecutionContext.Implicits.global
```

Comment:
Instead of depending on `global`, rely on an explicitly create threadpool which is appropriately sized.
Something like:

```
val threadPool = ThreadUtils.newDaemonFixedThreadPool(parallelism, "push-data-handler")
...

implicit val executionContext: ExecutionContext = ExecutionContext.fromExecutor(threadPool)
Future { ...

```

Reply:
@waitinfuture  do you think we need a separate threadpool for local write?

Reply:
originally, it was just part of the main thread, hence I had included it as a part of `global`

Reply:
You need a threadpool for the `Future` - the global threadpool gets used for a bunch of things, and if overloaded, it will result in not responding to clients on time: which is why a different threadpool.

In either case, we should not rely on `Implicits.global`